protocols/identify/src/identify.rs

```diff
@@ -758,4 +775,29 @@ mod tests {
 
         assert_eq!(connected_peer, swarm1_peer_id);
     }
+
+    #[test]
+    fn check_multiaddr_matches_peer_id() {
+        let peer_id = PeerId::random();
+        let other_peer_id = PeerId::random();
+        let mut addr: Multiaddr = ("/ip4/147.75.69.143/tcp/4001".to_string())
```

Comment:
```suggestion
        let mut addr: Multiaddr = ("/ip4/147.75.69.143/tcp/4001")
```

`.parse` should work on `&str` directly, no?

Reply:
yup! I was doing something else, and didn't rollback far enough 🤦